### PR TITLE
Arkit worldmap branch merge back into develop

### PR DESCRIPTION
**Pull request by blairmacintyre**
Cleaned up perferences so the io-app preferences and iOS system app preferences match.
Updated for iOS 12
Fixed issues with the AR tracking state messages (visible when URL bar visible) were sometimes incorrect
Added setWorldMap and getWorldMap commands for ARKit 2.0 shared maps.  getWorldMap is not enabled in shipping app, must be compiled by developers to create maps.

### CODE REVIEW

**Comment by davisjackie321** [comment]
Jkkk

**Comment by vincentfretin** [comment]
Hi @blairmacintyre I wanted to try your branch. I tried to build the app from your branch but the build fails with Swift 4.2 errors on PopupDialog and DynamicBlurView.
Does the build work on your machine?
I have latest XCode 10 and all updates on my iPad running iOS 12.
The app build successfully on master and develop. I'm running `pod install` after switching branch.

**Comment by blairmacintyre** [comment]
You need to update your pod installation.

Try "pod update" before "pod install" and see if that fixes it.

**Comment by blairmacintyre** [comment]
(also, I just committed some fixes, make sure you get them!)

**Comment by vincentfretin** [comment]
Thanks. Indeed, `pod update` fixed the Swift issues. I uncommented `#define ALLOW_GET_WORLDMAP` and built it. I tested your persistent example in webxr-polyfill develop, added a cube, put the app in background, and opened it again, relocalization was successful. Good work. ;-)

**Comment by blairmacintyre** [comment]
I've merged it over, and actually fixed some more bugs.

Please keep in mind that this is really unlikely to be standardized on any time soon, but it's certainly going to let us create some fun demos (that only work in WebXR Viewer, alas).

**Comment by blairmacintyre** [comment]
Let me know what you do! :)

File: XRViewer/ARKController/ARKController.h [hunk 1]
```diff
@@ -260,9 +289,41 @@ typedef void (^ActivateDetectionImageCompletionBlock)(BOOL success, NSString* er
...
+ 
+  - Fails if tracking isn't initialized, or if the acquisition of a World Map fails for some other reason
+ 
+  @param completion The completion block that will be called with the outcome of the acquisition of the world map
+  */
+- (void)getWorldMap:(GetWorldMapCompletionBlock)completion;
+
+/**
+ Set the current tracker World Map from a base64 encoded text string, probably passed in from Javascript.  Also saves this to
+ local storage.
```
Comment (vincentfretin) [inline XRViewer/ARKController/ARKController.h]: You wrote "Also saves this to local storage." here. Just to confirm, this is not currently true right?
`//[self _saveWorldMap:map];` is commented in `setWorldMap`
If I understood your code correctly, local storage here has no relation with the browser localStorage. local storage is just a file in a folder.
From javascript, if I want to keep the worldmap for a given site, I use getWorldMap and save it to localStorage, indexedDB or send it to a server. Then with the base64 string I retrieved from wherever I want, I use setWorldMap.
Do you have some number about the size in kB or MB that is needed to save this base64 string for a small to medium area? Well I'll do some tests anyway.
I understood you compress the worldmap with zlib then base64 it. Just for curiosity, did you test other compression algorithm or some options?

Comment (blairmacintyre) [reply XRViewer/ARKController/ARKController.h]: right, the web "setWorldMap" does not overwrite the stored map.  I need to change that comment.  Originally I also saved it to local storage, but when using the browser for development, this ended up being a bad choice.  Mostly, I wanted to scan a local room and save, and then no matter what else I did, I would like to be able to "reload".

Your workflow is correct.  We are planning on NOT shipping getWorldMap in the shipping browser as it feels like a big privacy hole.  The idea is that a developer would build the app with getWorldMap and create spaces they need;  the users of the app would end up downloading and calling setWorldMap as they use the app.

No reason I chose zlib:  that's what the sample code I had used.  The format is "easiest path to something."  This isn't really intended to be a feature that is advertised and widely used, I built it for some demos we want to do.

**Pull request merged**